### PR TITLE
CONSOLE-2477: Improve OperatorHub -> Sources view

### PR DESCRIPTION
**Pull request by harishsurf**
This change modifies operator-hub->Sources tab.
 Jira: https://issues.redhat.com/browse/CONSOLE-2477

The Sources tab list view now includes 2 new columns for the Catalog Sources:
**Status** and **Registry Poll Interval**

Kebab Action menu changes:
For **default sources:** Only the existing Disable and Edit CatalogSource 
For **custom sources:** The menus match the Actions menu in the Catalog Source

sources_tab

Signed-off-by: Harish <hgovinda@redhat.com>

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx [hunk 1]
```diff
@@ -249,10 +253,12 @@ export const CreateSubscriptionYAML: React.FC<CreateSubscriptionYAMLProps> = (pr
 
 const tableColumnClasses = [
   classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-xs-6'),
-  classNames('col-lg-2', 'col-md-3', 'col-sm-4', 'col-xs-6'),
-  classNames('col-lg-2', 'col-md-3', 'col-sm-4', 'hidden-xs'),
+  classNames('col-lg-1', 'col-md-2', 'col-sm-2', 'hidden-xs'),
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:256]: We should move away from the Bootstrap column classes while in the code. This lets us fit more columns in the table. See #3669 for an example.

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx [hunk 2]
```diff
@@ -348,33 +364,45 @@ const CatalogSourceList: React.FC<TableProps> = (props) => {
...
       },
       {
         title: t('catalog-source~Endpoint'),
         sortField: 'endpoint',
         transforms: [sortable],
-        props: { className: tableColumnClasses[3] },
+        props: { className: tableColumnClasses[4] },
+      },
+      {
+        title: t('catalog-source~Registry Poll Interval'),
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:392]: We're switching to sentence case for column headers.

```suggestion
        title: t('catalog-source~Registry poll interval'),
```

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx [hunk 3]
```diff
@@ -420,6 +448,12 @@ const DisabledPopover: React.FC<DisabledPopoverProps> = ({ operatorHub, sourceNa
   );
 };
 
+const getRegistryPollInterval = (catalogSource: CatalogSourceKind): string => {
+  return catalogSource.spec?.updateStrategy
+    ? catalogSource.spec.updateStrategy.registryPoll.interval
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:453]: Let's handle the case where `registryPoll` is missing

```suggestion
    ? catalogSource.spec.updateStrategy.registryPoll?.interval
```

**Review by spadgett — COMMENTED** [review]
Thanks @harishsurf

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx [hunk 4]
```diff
@@ -248,11 +252,13 @@ export const CreateSubscriptionYAML: React.FC<CreateSubscriptionYAMLProps> = (pr
 };
 
 const tableColumnClasses = [
-  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-xs-6'),
-  classNames('col-lg-2', 'col-md-3', 'col-sm-4', 'col-xs-6'),
-  classNames('col-lg-2', 'col-md-3', 'col-sm-4', 'hidden-xs'),
-  classNames('col-lg-3', 'col-md-3', 'hidden-sm', 'hidden-xs'),
-  classNames('col-lg-2', 'hidden-md', 'hidden-sm', 'hidden-xs'),
+  classNames('pf-m-visible-on-xs'),
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:255]: I think you could just leave the `pf-m-visible-on-xs` classname off.

```suggestion
  '',
```

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx [hunk 5]
```diff
@@ -248,11 +252,13 @@ export const CreateSubscriptionYAML: React.FC<CreateSubscriptionYAMLProps> = (pr
...
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+  classNames('pf-m-visible-on-xs'),
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:257]:
```suggestion
  '',
```

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx [hunk 6]
```diff
@@ -420,6 +448,12 @@ const DisabledPopover: React.FC<DisabledPopoverProps> = ({ operatorHub, sourceNa
   );
 };
 
+const getRegistryPollInterval = (catalogSource: CatalogSourceKind): string => {
+  return catalogSource.spec?.updateStrategy
+    ? catalogSource.spec.updateStrategy.registryPoll?.interval
+    : undefined;
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx:454]: Should have caught this on last review, but this is the same as

```suggestion
  return catalogSource.spec?.updateStrategy?.registryPoll?.interval;
```

File: frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx (hunk 4 above)
Comment (harishsurf) [reply frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx]: yep, this works

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for approvals
/assign @yapei @ahardin-rh @sferich888

**Comment by yapei** [comment]
QE added test case and tested locally on a cluster-bot cluster, no issues found, detail verification steps can be found from test case

**Comment by spadgett** [comment]
Removing the hold since we're below our SLO.

/hold cancel
/retest

**Pull request merged**